### PR TITLE
Implement CSI clone

### PR DESCRIPTION
**Pull request by awels**
Signed-off-by: Alexander Wels <awels@redhat.com>

**What this PR does / why we need it**:
Implemented support for CSI clone. The 'clone' is a simple cp command which uses reflink if on an XFS filesystem, other
wise it falls back to a regular copy.

Verify that the source is on the same node as the target otherwise return an error to the caller. This ensures the pod is scheduled on the proper node.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Enhancement: Support for CSI clone on same node.
```

### CODE REVIEW

**Comment by mhenriks** [comment]
looks good just missing functional tests

**Comment by awels** [comment]
yeah, some of those functional tests are in the kubernetes storage tests themselves. But as you can see they are failing on the capacity check. I will have to disable those tests for now in the kubernetes tests, and add some equivalent ones in the regular functional tests.

File: pkg/hostpath/controllerserver.go [hunk 1]
```diff
@@ -119,7 +156,31 @@ func (hpc *hostPathController) CreateVolume(ctx context.Context, req *csi.Create
 			return nil, fmt.Errorf("failed to create volume %v: %w", req.GetName(), err)
 		}
 		klog.V(4).Infof("created volume %s at path %s", req.GetName(), filepath.Join(hpc.cfg.StoragePoolDataDir[storagePoolName], req.GetName()))
-	}	
+	}
+
+	if req.GetVolumeContentSource() != nil {
```
Comment (akalenyu) [inline pkg/hostpath/controllerserver.go:161]: If you didn't mean to call it twice, `if source := req.GetVolumeContentSource(); source != nil {` could work here
(I've seen two or more calls to this from a single func throughout the PR so might be missing something)

**Pull request closed without merging**